### PR TITLE
fixes more clippy warnings

### PR DESCRIPTION
**Pull request by jimmyhmiller**

### CODE REVIEW

**Comment by jimmyhmiller** [comment]
Messed up the macro change. Going to revert that one. It was the most iffy of them all

File: yjit/src/cruby.rs [hunk 1]
```diff
@@ -96,6 +96,7 @@ pub type size_t = u64;
 pub type RedefinitionFlag = u32;
 
 #[allow(dead_code)]
+#[allow(clippy::useless_transmute)]
```
Comment (maximecb) [inline yjit/src/cruby.rs:99]: lol

File: yjit/src/utils.rs [hunk 2]
```diff
@@ -122,12 +122,20 @@ yjit_print_iseq(const rb_iseq_t *iseq)
 
 #[cfg(target_arch = "aarch64")]
 macro_rules! c_callable {
-    (fn $f:ident $args:tt $(-> $ret:ty)? $body:block) => { extern "C" fn $f $args $(-> $ret)? $body };
+    ($(#[$outer:meta])*
```
Comment (maximecb) [inline yjit/src/utils.rs:125]: What's outer:meta ???

**Review by maximecb — APPROVED** [review]
Looks good

File: yjit/src/utils.rs (hunk 2 above)
Comment (XrXr) [reply yjit/src/utils.rs:125]: The doc comments outside of functions like `///` are actually sugar for meta attributes like `#[doc="..."]` and this is how you'd match against them.

**Pull request merged**